cvmfs/sync_item.cc

```diff
@@ -10,11 +10,77 @@
 
 using namespace std;  // NOLINT
 
+
+static bool ProcessCatalogProperties(const std::string &scratch_path, bool &external_data) {
+  char buf[128];
+  external_data = false;
+  ssize_t retval;
+  if (-1 != (retval = lgetxattr(scratch_path.c_str(), "user.external_data", buf, 127))) {
```

Comment:
`lgetxattr` is encapsulated in platform.h as `platform_lgetxattr`.  In platform.h (platform_linux.h, platform_osx.h) there are a number of routines that work differently on Mac and Linux.